controllers/post.js

```diff
@@ -1,8 +1,15 @@
 const Post = require("../models/post");
 
-const getAllPosts = async (req, res) => {
+// if sender provided, returns all the sender's posts
+// otherwise, return all posts
+const getPosts = async (req, res) => {
+  const { sender } = req.query;
+  let posts = [];
```

Comment:
You don't need this parameter outside the try-catch block, move it one line below